Parsers/ASimNetworkSession/Tests/SonicWall_SonicOS_ASimNetworkSession_SchemaTest.csv

```diff
@@ -0,0 +1,301 @@
+"(2) Info: extra unnormalized column [OldFileName]"
+"(2) Info: extra unnormalized column [OldFilePath]"
+"(2) Info: extra unnormalized column [OldFilePermission]"
+"(2) Info: extra unnormalized column [OldFileSize]"
+"(2) Info: extra unnormalized column [OldFileType]"
+"(2) Info: extra unnormalized column [OriginalLogSeverity]"
+"(2) Info: extra unnormalized column [OutboundInterface]"
+"(2) Info: extra unnormalized column [ProcessID]"
+"(2) Info: extra unnormalized column [ProcessName]"
+"(2) Info: extra unnormalized column [Protocol]"
+"(2) Info: extra unnormalized column [ReceiptTime]"
+"(2) Info: extra unnormalized column [ReceivedBytes]"
+"(2) Info: extra unnormalized column [ReceivedPackets]"
+"(2) Info: extra unnormalized column [ReceivedPackets_Field]"
+"(2) Info: extra unnormalized column [RemoteIP]"
+"(2) Info: extra unnormalized column [RemotePort]"
+"(2) Info: extra unnormalized column [ReportReferenceLink]"
+"(2) Info: extra unnormalized column [RequestContext]"
+"(2) Info: extra unnormalized column [RequestCookies]"
+"(2) Info: extra unnormalized column [RequestURL]"
+"(2) Info: extra unnormalized column [RequestURL_]"
+"(2) Info: extra unnormalized column [RuleDetails]"
+"(2) Info: extra unnormalized column [SentBytes]"
+"(2) Info: extra unnormalized column [SentPackets]"
+"(2) Info: extra unnormalized column [SentPackets_Field]"
+"(2) Info: extra unnormalized column [SerialNumber]"
+"(2) Info: extra unnormalized column [ServiceName]"
+"(2) Info: extra unnormalized column [SimplifiedDeviceAction]"
+"(2) Info: extra unnormalized column [SoleraNPCSURL]"
+"(2) Info: extra unnormalized column [SourceDnsDomain]"
+"(2) Info: extra unnormalized column [SourceHostName]"
+"(2) Info: extra unnormalized column [SourceIP]"
+"(2) Info: extra unnormalized column [SourceIPv6Address]"
+"(2) Info: extra unnormalized column [SourceNTDomain]"
+"(2) Info: extra unnormalized column [SourcePortSPT]"
+"(2) Info: extra unnormalized column [SourcePort]"
+"(2) Info: extra unnormalized column [SourceProcessId]"
+"(2) Info: extra unnormalized column [SourceProcessName]"
+"(2) Info: extra unnormalized column [SourceServiceName]"
+"(2) Info: extra unnormalized column [SourceSystem]"
+"(2) Info: extra unnormalized column [SourceUserID]"
+"(2) Info: extra unnormalized column [SourceUserPrivileges]"
+"(2) Info: extra unnormalized column [SourceVPNPolicyName]"
+"(2) Info: extra unnormalized column [SourceZone]"
+"(2) Info: extra unnormalized column [StartTime]"
+"(2) Info: extra unnormalized column [TenantId]"
+"(2) Info: extra unnormalized column [ThreatDescription]"
+"(2) Info: extra unnormalized column [ThreatSeverity]"
+"(2) Info: extra unnormalized column [URLPathName]"
+"(2) Info: extra unnormalized column [UrlCategory]"
+"(2) Info: extra unnormalized column [UserSessionDuration]"
+"(2) Info: extra unnormalized column [UserSessionType]"
+"(2) Info: extra unnormalized column [_ItemId]"
+"(2) Info: extra unnormalized column [fileid]"

```

Comment:
@jaimeesc 
Please remove fields that are not matching with NetworkSession schema. If you are getting many outside of NetworkSession fields, it might be because we are allowing log type other than networksession in the parser. Please set appropriate filter condition to only let take NetworkSession traffic.
What is NetworkSession traffic - https://learn.microsoft.com/en-us/azure/sentinel/normalization-schema-network#schema-overview

Reply:
Please only map with fields that are present in NetworkSession schema here: https://learn.microsoft.com/en-us/azure/sentinel/normalization-schema-network#schema-overview

If see any field thats relevant for NetworkSession traffic but not in schema, please add to [AdditionalFields](https://learn.microsoft.com/en-us/azure/sentinel/normalization-common-fields#additionalfields)

Reply:
also we need to do project-away on fields not matching with NetworkSession

Reply:
There are many fields added for enrichment and friendliness, as well as fields renamed from fields with generic names that are not simple to understand based on a numeric value. For example I renamed "DeviceCustomString6Label" to "UserSessionDuration" thinking that was an okay thing to do.

I've read those pages and still do not understand what constitutes a network session and what doesn't, if not the presence of src/dst IP address.

So to resolve this, I either have to: not provide friendly field names (keep the default names), map them to supported and relevant field names if one is available, or use "AdditionalFields" to store the field and value as a name value pair?

Reply:
I see that I can use AdditionalFields to include additional details. I'm making the necessary changes to remove anything not in schema.

I'm still not understanding what constitutes a network session and what doesn't, if not the presence of src/dst IP address. If you could please clearly explain what filters should be applied or what components of the log are used to determine if an event is a network session.

Thanks for any insight you can provide.

Reply:
answered in a different comment above.